gateway/Dockerfile

```
@@ -1,17 +1,10 @@
 #
 # === Build image ===
 #
-FROM rust:1.80-alpine3.20 AS chef
-
-# Patching musl to use mimalloc, see docker/README.md.
-# Source: https://github.com/tweag/rust-alpine-mimalloc
-COPY docker/build.sh docker/mimalloc.diff /tmp/
-RUN /tmp/build.sh
-ENV LD_PRELOAD=/usr/lib/libmimalloc.so
-ENV RUSTFLAGS="-C target-feature=+crt-static"
+FROM rust:1.82-bookworm AS chef
 
 COPY rust-toolchain.toml rust-toolchain.toml
-RUN apk add --no-cache musl-dev && cargo install --target "$(rustc -vV | sed -n "s|host: ||p")" cargo-chef
+RUN cargo install --target "$(rustc -vV | sed -n "s|host: ||p")" cargo-chef
```

Is this `--target` even needed now?  Would have thought not.

```suggestion
RUN cargo install cargo-chef
```